src/rgw/rgw_auth.cc

```diff
@@ -91,6 +91,29 @@ transform_old_authinfo(const req_state* const s)
                                  s->system_request));
 }
 
+vector<std::string> parse_auth_order(const CephContext* cct)
+{
+  vector <std::string> result;
+
+  /* TODO: decide on the names here, we're starting with external, local though
+     we could be more specific as keystone, ldap, rados etc or even more
+     specific wherein they map to the name of the class so a `get_name` ought to
+     work. The derived classes then need to create the strategy map mapping
+     their types to these allowed verbs */
+  const std::set <std::string_view> allowed_auth = { "external", "local" };
+  std::vector <std::string> default_order = { "external", "local"};
+  // supplied strings may contain a space, so let's bypass that
+  boost::split(result, cct->_conf->rgw_auth_order,
+               boost::is_any_of(", "), boost::token_compress_on);
+  for (const auto &it : result){
+    if (allowed_auth.find(it) == allowed_auth.end()) {
+      dout(0) << __func__ << "couldn't find " << it << " auth method using defaults" << dendl;
```

Comment:
Maybe `ldout` instead?

Reply:
ldout requires non const ctx, which we don't have (as the AWSAuth init fn also takes a const ptr to cct)